python/tskit/intervals.py

```diff
@@ -0,0 +1,600 @@
+#
+# MIT License
+#
```

Comment:
We need to keep the original copyright notices I'm afraid.

I think it might be easier to start this with a new branch:

- first copy in the msprime code as-is, with no changes. Make a commit, with the message including text from contributors like we have 
- Next update the utils code as required, and commit
- Then make the changes to intervals.py and tests needed to bring them into tskit, and commit. This includes new copyright headers *in addition to* the existing ones.

Sorry I know this is tedious, but licencing issues always are and we really don't want there to be any grey areas with it.

Reply:
BTW, if you are happy to do it, and it's easy, go ahead @jeromekelleher . That way it'll be sure to be done as you want.

Reply:
Leave it with me

Reply:
Great, thanks